### PR TITLE
implement focusRequirements field

### PR DESCRIPTION
**Pull request by lhorie**
Adds a focusRequirements field to manifest.json, which forcefully installs specified packages. This is useful when there are commonly used tools that aren't part of a project's build graph, for example codemodding tools. See README.md for details

### CODE REVIEW

File: commands/focus.js [hunk 1]
```diff
@@ -96,7 +99,8 @@ const focus /*: Focus */ = async ({
 
   const env = process.env;
   const path = dirname(node) + ':' + String(process.env.PATH);
-  const spawnArgs = [yarn, 'workspaces', 'focus', ...args];
+  const focusable = [...new Set([...packages, ...focusRequirements])];
+  const spawnArgs = [yarn, 'workspaces', 'focus', ...focusable];
```
Comment (lhorie) [inline commands/focus.js:103]: this is the meat and potatoes of the change

File: commands/focus.js [hunk 2]
```diff
@@ -42,11 +42,14 @@ const focus /*: Focus */ = async ({
     hooks = {},
     workspace,
     dependencySyncRule,
+    focusRequirements = [],
```
Comment (lhorie) [inline commands/focus.js:45]: spread the new field out of manifest.json

File: commands/focus.js (hunk 1 above)
Comment (lhorie) [inline commands/focus.js:103]: then merge it here

**Comment by lhorie** [comment]
rest of changes are type fixes, some other random cleanup and some documentation fixes

**Review by AlexMSmithCA — APPROVED** [review]

**Pull request merged**